### PR TITLE
Updates from virtualenv test with MacOS and ocp4-cluster

### LINKED ISSUE #182: MacOS launching 4.x cluster Python crash: TASK [infra-aws-capacity-reservation : Create on-demand capacity reservations and save result]
Saw below today on MacOS, using python virtualenv with https://github.com/konveyor/mig-agnosticd/blob/master/requirements.txt

```
TASK [infra-aws-capacity-reservation : Create on-demand capacity reservations and save result] ******************************************************************************
Wednesday 20 October 2021  13:54:15 -0400 (0:00:00.196)       0:00:12.054 ***** 
objc[11200]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[11200]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
ERROR! A worker was found in a dead state
```

Seems odd, but it looks like when I run this my system Python is being run and crashes.  I would have expected to see the Python from my virtualenv

```
$ which python
/Users/jmatthews/git/jwmatthews/mig-agnosticd/env/bin/python

$ which ansible-playbook
/Users/jmatthews/git/jwmatthews/mig-agnosticd/env/bin/ansible-playbook
```
```
Process:               Python [11200]
Path:                  /usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python
Identifier:            Python
Version:               3.9.7 (3.9.7)
Code Type:             X86-64 (Native)
Parent Process:        Python [10625]
Responsible:           Terminal [611]
User ID:               501

Date/Time:             2021-10-20 13:54:15.571 -0400
OS Version:            Mac OS X 10.15.7 (19H1419)
Report Version:        12
Bridge OS Version:     5.5 (18P4759a)
Anonymous UUID:        5896D922-FB03-C409-AC7F-3C2F23BD3476


Time Awake Since Boot: 1600000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace OBJC, Code 0x1

Application Specific Information:
objc[11200]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
+[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
crashed on child side of fork pre-exec

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff7286ead6 __abort_with_payload + 10
1   libsystem_kernel.dylib        	0x00007fff728703df abort_with_payload_wrapper_internal + 80
2   libsystem_kernel.dylib        	0x00007fff7287038f abort_with_reason + 19
3   libobjc.A.dylib               	0x00007fff71576820 _objc_fatalv(unsigned long long, unsigned long long, char const*, __va_list_tag*) + 114
4   libobjc.A.dylib               	0x00007fff715767ae _objc_fatal(char const*, ...) + 135
5   libobjc.A.dylib               	0x00007fff71567a7c performForkChildInitialize(objc_class*, objc_class*) + 318
6   libobjc.A.dylib               	0x00007fff7156802f initializeNonMetaClass + 527
7   libobjc.A.dylib               	0x00007fff71569062 initializeAndMaybeRelock(objc_class*, objc_object*, mutex_tt<false>&, bool) + 214
8   libobjc.A.dylib               	0x00007fff71559d45 lookUpImpOrForward + 1072
9   libobjc.A.dylib               	0x00007fff71559399 _objc_msgSend_uncached + 73
10  com.apple.CoreFoundation      	0x00007fff385e429d -[__NSDictionaryM __setObject:forKey:] + 443
11  com.apple.CoreFoundation      	0x00007fff385e161d parseXMLElement + 2754
12  com.apple.CoreFoundation      	0x00007fff385e0f8b parseXMLElement + 1072
13  com.apple.CoreFoundation      	0x00007fff385e0007 _CFPropertyListCreateFromUTF8Data + 1880
14  com.apple.CoreFoundation      	0x00007fff386e5f19 _CFPropertyListCreateWithData + 813
15  com.apple.CoreFoundation      	0x00007fff385df317 CFPropertyListCreateWithData + 51
16  com.apple.CoreFoundation      	0x00007fff385de246 _CFBundleCopyInfoDictionaryInDirectoryWithVersion + 814
17  com.apple.CoreFoundation      	0x00007fff385ddbbd _CFBundleRefreshInfoDictionaryAlreadyLocked + 111
18  com.apple.CoreFoundation      	0x00007fff385ddb3c CFBundleGetInfoDictionary + 33
19  com.apple.CoreFoundation      	0x00007fff386e55af _CFBundleCreate + 715
20  com.apple.CoreFoundation      	0x00007fff385ce82e CFBundleGetMainBundle + 148
21  com.apple.CoreFoundation      	0x00007fff385f00cd _CFPrefsGetCacheStringForBundleID + 71
22  com.apple.CoreFoundation      	0x00007fff385f6b88 -[CFPrefsPlistSource setDomainIdentifier:] + 92
23  com.apple.CoreFoundation      	0x00007fff385f6aba -[CFPrefsPlistSource initWithDomain:user:byHost:containerPath:containingPreferences:] + 115
24  com.apple.CoreFoundation      	0x00007fff385f6a2c -[CFPrefsManagedSource initWithDomain:user:byHost:containingPreferences:] + 62
25  com.apple.CoreFoundation      	0x00007fff385f69bd __85-[_CFXPreferences(PlistSourceAdditions) withManagedSourceForIdentifier:user:perform:]_block_invoke + 108
26  com.apple.CoreFoundation      	0x00007fff385f6919 -[_CFXPreferences(SourceAdditions) withSources:] + 56
27  com.apple.CoreFoundation      	0x00007fff385f1bf7 -[_CFXPreferences(PlistSourceAdditions) withManagedSourceForIdentifier:user:perform:] + 253
28  com.apple.CoreFoundation      	0x00007fff385f1adf -[CFPrefsSearchListSource addManagedSourceForIdentifier:user:] + 101
29  com.apple.CoreFoundation      	0x00007fff3874b6e2 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke.249 + 298
30  com.apple.CoreFoundation      	0x00007fff385f1951 -[_CFXPreferences(SearchListAdditions) withSearchLists:] + 56
31  com.apple.CoreFoundation      	0x00007fff385f17f5 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 299
32  com.apple.CoreFoundation      	0x00007fff385f133a -[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 364
33  com.apple.CoreFoundation      	0x00007fff385f10c4 -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 145
34  com.apple.CoreFoundation      	0x00007fff385f0ff7 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 101
35  com.apple.SystemConfiguration 	0x00007fff45ff487b SCDynamicStoreCopyProxiesWithOptions + 155
36  _scproxy.cpython-39-darwin.so 	0x000000010ce6b8cd get_proxy_settings + 32
37  org.python.python             	0x000000010a3f080d cfunction_vectorcall_NOARGS + 104

```

### LINKED ISSUE #183: Need to update requirements.txt:  crypt.crypt not supported on Mac OS X/Darwin, install passlib python module
Related to ocp4.x provision on MacOS

```
TASK [bastion-student-user : Create user] ***********************************************************************************************************************************
task path: /Users/jmatthews/git/jwmatthews/agnosticd/ansible/roles/bastion-student-user/tasks/main.yml:25
Wednesday 20 October 2021  15:19:50 -0400 (0:00:00.036)       0:08:09.010 ***** 
fatal: [bastion.jwm1020ocp4a.internal]: FAILED! => {
    "msg": "crypt.crypt not supported on Mac OS X/Darwin, install passlib python module. crypt.crypt not supported on Mac OS X/Darwin, install passlib python module"
}
```

Will update requirements.txt shortly after I've completed a successful test

### LINKED ISSUE #184: delete_ocp4_workshop.sh: Fails  "The error was: 'dict object' has no attribute 'creation_time"
On MacOS, running in a virtualenv

```
TASK [Grab and set stack creation_time] *************************************************************************************************************************************
Wednesday 20 October 2021  15:25:59 -0400 (0:00:01.315)       0:00:12.885 ***** 
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'creation_time'\n\nThe error appears to be in '/Users/jmatthews/git/jwmatthews/agnosticd/ansible/configs/ocp4-cluster/destroy_env_ec2.yml': line 25, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: Grab and set stack creation_time\n    ^ here\n"}

```

Believe above is related to info @jmontleon shared in the past on a change of behavior with Ansible versions
https://github.com/ansible-collections/amazon.aws/issues/193

### PR DESCRIPTION
**Pull request by jwmatthews**
Fixes #182 
Fixes #183 
Fixes #184 

Related to below JIRAs:
* https://issues.redhat.com/browse/MIG-848
* https://issues.redhat.com/browse/MIG-841

### CODE REVIEW

File: 4.x/create_ocp4_workshop.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+# Required for MacOS with virtualenv
+# as per https://github.com/konveyor/mig-agnosticd/issues/182
+export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
```
Comment (jwmatthews) [inline 4.x/create_ocp4_workshop.sh:5]: Spoke to @weshayutin and I'll add a check in to only do this on uname=Darwin

File: 4.x/delete_ocp4_workshop.sh [hunk 2]
```diff
@@ -1,5 +1,9 @@
 OUR_DIR=`pwd`
 
+# Required for MacOS with virtualenv
+# as per https://github.com/konveyor/mig-agnosticd/issues/182
+export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
```
Comment (jwmatthews) [inline 4.x/delete_ocp4_workshop.sh:5]: Spoke to @weshayutin and I'll add a check in to only do this on uname=Darwin

**Comment by jwmatthews** [comment]
Added a check for create/delete of OCP4 on "Darwin".
Ran through and looking good on MacOS test.

Good to go on my side.

**Review by weshayutin — APPROVED** [review]
Just because I love you and I feel compelled to give you shit.. your indent is 2 spaces, the other bits of the file indent 4 spaces.  +2 /me runs :)

**Pull request merged**